src/helpers/updateContact.spec.js

```diff
@@ -0,0 +1,66 @@
+import {
+  updateContact,
+  getContactsToUpdate,
+  getLastSuccessOfService
+} from './updateContact'
+import { johnDoeContact } from './testData'
+
+import CozyClient from 'cozy-client'
+const client = new CozyClient({})
+
+describe('checkAndUpdate', () => {
+  it('should returns contact with new attributes', () => {
+    const expected = {
+      ...johnDoeContact,
+      displayName: 'John Doe',
+      fullname: 'John Doe',
+      indexes: {
+        byFamilyNameGivenNameEmailCozyUrl:
+          'DoeJohnjohn.doe@cozycloud.ccjohndoe.mycozy.cloud'
+      }
+    }
+    expect(updateContact(johnDoeContact)).toEqual(expected)
+  })
+})
+
+describe('getContactsToUpdate', () => {
+  it('should returns only contact to update', async () => {
+    const lastSuccess = new Date()
+    const contactToUpdate = johnDoeContact
+    const contactUpToDate = updateContact(johnDoeContact)
+    client.queryAll = jest.fn(() =>
+      Promise.resolve([contactToUpdate, contactUpToDate])
+    )
+    const expected = [contactToUpdate]
+    expect(await getContactsToUpdate(client, lastSuccess)).toEqual(expected)
+  })
+})
+
+describe('getLastSuccessOfService', () => {
+  it('should do something', async () => {
+    const triggersResult = {
+      data: [
+        {
+          id: '6e0847f496cbeb11e3ed653f170086bd'
+        }
+      ]
+    }
+    const trigger = {
+      data: {
+        current_state: {
+          last_success: '2020-08-12T11:14:46.060006+02:00'
+        }
+      }
+    }
+
+    client.query = jest
+      .fn()
+      .mockReturnValueOnce(triggersResult)
+      .mockReturnValueOnce(trigger)
+
+    const expected = '2020-08-12T11:14:46.060006+02:00'
+    expect(await getLastSuccessOfService(client, 'serviceName')).toMatch(
+      expected
+    )
+  })
+})
```

Comment:
Nit: Personally, I think it's better to test the complete service, instead of testing each and each function, that could be thought of as implementation details.

Reply:
Ok i'll do this way next time

Reply:
Ah, it's a good point 👍